pkg/analysis/passes/metadatapaths/metadatapaths.go

```diff
@@ -10,75 +10,89 @@ import (
 	"github.com/grafana/plugin-validator/pkg/analysis/passes/archive"
 	"github.com/grafana/plugin-validator/pkg/analysis/passes/logos"
-	"github.com/grafana/plugin-validator/pkg/analysis/passes/metadata"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/nestedmetadata"
 	"github.com/grafana/plugin-validator/pkg/analysis/passes/screenshots"
 )
 
 var (
 	invalidPath            = &analysis.Rule{Name: "invalid-path", Severity: analysis.Error}
-	pathRelativeToMetadata = &analysis.Rule{Name: "path-relative-to-metadata", Severity: analysis.Error}
-	invalidRelativePath    = &analysis.Rule{Name: "invalid-relative-path", Severity: analysis.Error}
-	pathNotExists          = &analysis.Rule{Name: "path-not-exists", Severity: analysis.Error}
+	pathRelativeToMetadata = &analysis.Rule{
+		Name:     "path-relative-to-metadata",
+		Severity: analysis.Error,
+	}
+	invalidRelativePath = &analysis.Rule{Name: "invalid-relative-path", Severity: analysis.Error}
+	pathNotExists       = &analysis.Rule{Name: "path-not-exists", Severity: analysis.Error}
 )
 
 var Analyzer = &analysis.Analyzer{
-	Name:     "metadatapaths",
-	Run:      checkMetadataPaths,
-	Requires: []*analysis.Analyzer{screenshots.Analyzer, logos.Analyzer},
-	Rules:    []*analysis.Rule{invalidPath, pathRelativeToMetadata, invalidRelativePath, pathNotExists},
+	Name: "metadatapaths",
+	Run:  checkMetadataPaths,
+	Requires: []*analysis.Analyzer{
+		screenshots.Analyzer,
+		logos.Analyzer,
+		nestedmetadata.Analyzer,
+		archive.Analyzer,
+	},
+	Rules: []*analysis.Rule{
+		invalidPath,
+		pathRelativeToMetadata,
+		invalidRelativePath,
+		pathNotExists,
+	},
+}
+
+type CheckPath struct {
+	kind           string
+	path           string
+	relativeToPath string
 }
 
 func checkMetadataPaths(pass *analysis.Pass) (interface{}, error) {
-	var paths []struct {
-		kind string
-		path string
-	}
+	var paths []CheckPath
 
-	// Screenshots
-	screenshots, ok := pass.ResultOf[screenshots.Analyzer].([]metadata.MetadataScreenshots)
-	// Will be nil if no screenshots were found.
-	if ok {
-		for _, s := range screenshots {
-			paths = append(paths, struct {
-				kind string
-				path string
-			}{
-				kind: "screenshot",
-				path: s.Path,
-			})
-		}
+	archiveDir, ok := pass.ResultOf[archive.Analyzer].(string)
+	if !ok {
+		return nil, nil
 	}
 
-	// Logos
-	logos, ok := pass.ResultOf[logos.Analyzer].(metadata.MetadataLogos)
+	metadatamap, ok := pass.ResultOf[nestedmetadata.Analyzer].(nestedmetadata.Metadatamap)
 	if !ok {
 		return nil, nil
 	}
-	paths = append(paths, struct {
-		kind string
-		path string
-	}{
-		kind: "small logo",
-		path: logos.Small,
-	})
-	paths = append(paths, struct {
-		kind string
-		path string
-	}{
-		kind: "large logo",
-		path: logos.Large,
-	})
-
-	archiveDir := ""
-	archiveAnalyserResult := pass.ResultOf[archive.Analyzer]
-	if archiveAnalyserResult != nil {
-		archiveDir = archiveAnalyserResult.(string)
+
+	for currentPluginJson, meta := range metadatamap {
```

Comment:
similar to the logos case. the checks remain the same but they are now inside a loop of the metadatamap.

Also because this particular validator also checks for relative paths for images and screenshots, it has a "relativeTo" string which is where the images/screenshots in the plugin.json are relative to. 

The `key` of the metadatamap is the path to the pluginjson relative to the archivedir root folder (plugin.json for the main plugin json file for example)